File: python/src/multiagent_boilerplate/tools/builtin/read_artifact.py
from __future__ import annotations

from ..types import ToolContext


class ReadArtifactTool:
    name = "read_artifact"
    description = "Fetch the full content of a previously stored artifact by id, when a distilled summary isn't enough."
    input_schema = {"type": "object", "required": ["artifactId"], "properties": {"artifactId": {"type": "string"}}}
    exposable = False

    async def execute(self, input_: dict, ctx: ToolContext) -> dict:
        return await ctx.runtime.read_artifact(input_["artifactId"])


read_artifact_tool = ReadArtifactTool()

File: typescript/src/tools/builtin/readArtifact.ts
import type { Tool } from "../types.js";

export const readArtifactTool: Tool<{ artifactId: string }, unknown> = {
  name: "read_artifact",
  description: "Fetch the full content of a previously stored artifact by id, when a distilled summary isn't enough.",
  inputSchema: {
    type: "object",
    required: ["artifactId"],
    properties: { artifactId: { type: "string" } },
  },
  async execute(input, ctx) {
    return ctx.runtime.readArtifact(input.artifactId);
  },
};
